Purchase/index.ts

```diff
@@ -134,8 +134,10 @@ export namespace Purchase {
 		])
 	}
 
+
+	
 	export async function compileExpense(
-		compileData: { purchases: { buyer: string; purpose: string; amount: Amount }[] },
+		compileData: Record<string, { purpose: string; date: isoly.Date; amount: Amount }[]>,
```

Comment:
do we maybe want to extract these types to some Creatable of something? they are used in the API quite a lot as well if i am not mistaken?